### ISSUE TITLE
Client-side validation and layered conditions preventing submission

### ISSUE THREAD

**Issue by jmauzyk** [verbb/formie#726]
**Description**

When the Validate Form on Submit setting is enabled, it prevents form submission in certain situations where a group has one set of conditions and its fields has another set of conditions. Here's an example:

**Steps to reproduce**
1. Create a required dropdown field with options Sales, Marketing, and Operations in addition to the default Select an option.
2. Create a group field and set a condition to show when the dropdown is set to Sales or Marketing.
3. Create two required single-line text fields within the group. Set a condition on one to hide when the dropdown is set to Marketing.
4. Set the Validate Form on Submit form setting to true.
5. From the front end, select Operations in the dropdown and submit the form. It should return a error message "Couldn’t save submission due to errors." Even though the group is hidden, the required field with conditions isn't explicitly hidden and seems to be getting validated on submit. You should be able to see this if you switch the dropdown to Sales after attempting to submit.

### Form settings
- Multi-page form: No
- Submission Method: Page Reload
- Client-side Validation: Yes
- Custom Form Templates: No

**Additional info**
- Plugin version: 1.5.7
- Craft version: Pro 3.7.30
- Multi-site: No

**Comment by engram-design** [comment]
Yep, this is a tricky one because the two rules are cancelling each other out. This can be seen on initial load before and option has been picked from the top-level dropdown.

For the group field it'll evaluate that the field should be hidden. It'll also set any inner `input`, `textarea` or `select` elements to disabled (preventing client-side validation or sending values to the server). This would be expected.

Then, continuing down the conditions rulesets, the dropdown inside the Group field it evaluates to false (because the value isn't "Marketing"), so the field is shown, and marked as non-disabled. It's also required per your field settings.

So now we have a required dropdown field within a hidden group field that's triggering validation.

I'm going to have to add some handling to make inner fields "group-aware" as the inner conditions shouldn't even really be evaluating unless the group field's conditions evaluate to true.

**Comment by engram-design** [comment]
Fixed for the next release. To get this fix early, change your `verbb/formie` requirement in `composer.json` to:

```
"require": {
  "verbb/formie": "dev-craft-3 as 1.5.7",
  "...": "..."
}
```

Then run `composer update`.

**Comment by engram-design** [comment]
Fixed in [1.5.8](https://github.com/verbb/formie/releases/tag/1.5.8)
